### ISSUE TITLE
Direct challenge (reseted) generated the pool as outer join, not inner join

### ISSUE THREAD

**Issue by tomguta** [Razviar/mtga-pro-tracker#443]
**Describe the bug**
Direct challenge (reseted) generated a pool, that's not usable by both players.

**To Reproduce**
1. Not sure about this, but I reseted a game and changed a number of boosters couple of times.
2. The generated pool was identical for both players, not sure whether that's intentional.
3. The other player doesn't have all the cards in the pool.
4. Se attached:

**Expected behavior**
Inner join is used instead.

**Screenshots**
![image](https://user-images.githubusercontent.com/1763443/60133776-59e02700-979e-11e9-90e6-69816d5796ab.png)
![image](https://user-images.githubusercontent.com/1763443/60133783-61073500-979e-11e9-930e-9103cc48d02b.png)
Visible here:
https://mtgarena.pro/deckbuilder/?Magesa&pool-1561404265

**Comment by F43nd1r** [comment]
I can confirm this issue exists - and it makes the tourney mode unusable.

**Comment by zass30** [comment]
Confirming independently. I was going to open a new issue for this. Here are my repro steps:

1. Create a 2 person event with 9 boosters.
![image](https://user-images.githubusercontent.com/2244451/61003645-447c0880-a319-11e9-854d-ff3d6fe7fe65.png)

Note incubation druid is one of the rares.

2. Note that both players have the same identical card pool. Each player is supposed to get their own card pool.

3. One of the players doesn't own incubation druid:

![image](https://user-images.githubusercontent.com/2244451/61003873-9b81dd80-a319-11e9-9851-77688263b31f.png)


As the OP wrote, it appears there are two bugs:
a) outer join is used instead of inner join for the card set
b) boosters are duplicated for the players instead of independently generated in the pool

**Comment by Razviar** [comment]
It's fixed

**Comment by zass30** [comment]
Unable to verify. The direct limited lobby just shows a spinning wheel, and cannot create a new event:
![image](https://user-images.githubusercontent.com/2244451/61472708-dbf4e300-a939-11e9-9acc-1bba9465590d.png)

**Comment by Razviar** [comment]
Endless load fixed

**Comment by zass30** [comment]
Outer join not fixed. And duplicate decks not fixed either. Opponent doesn't own this card:

![image](https://user-images.githubusercontent.com/2244451/61849725-0b8a7a80-ae67-11e9-8e29-34682373d226.png)

**Comment by gbeck42** [comment]
Still waiting on fixed to duplicate decks and outer join. Is there any progress on this? I've got some friends who would love to mess around with this.

**Comment by Razviar** [comment]
Yep, this is on the list. But since all tracking was messed up after latest update, we had urgent issues to deal with first. I hope to fix this by the end of weekend.

**Comment by zass30** [comment]
any update?

**Comment by Razviar** [comment]
> any update?

It's under construction, unfortunately no updates or ETA, sorry. I'll let you know when we'll have something.

**Comment by zass30** [comment]
pinging on this. I think this is probably a relatively simple fix, and would fix the currently broken feature. It's a good cost/benefit to fix.

**Comment by Razviar** [comment]
Patches have been applied for both issues, please test.

**Comment by gbeck42** [comment]
I just tested with a friend of mine. Cards do not show on the initial preview screen and the build screen is empty after a quarter of a second after opening.

**Comment by zass30** [comment]
Seems to be an issue now with the count of cards. See game "rcfc"

In the direct challenge mode, the cards pool list shows 2x ravenous chucapabra, but when I go to the deck builder it only shows one:

![image](https://user-images.githubusercontent.com/2244451/66710638-7e521600-ed31-11e9-9e71-9a4ec50e611a.png)
![image](https://user-images.githubusercontent.com/2244451/66710640-81e59d00-ed31-11e9-8b9b-b31f8aaa08ec.png)

**Comment by zass30** [comment]
confirmed not fixed. See game rcfc now, bothplayers have same card pool still.

**Comment by Razviar** [comment]
Another patch applied
